invenio_notifications/backends/utils/loaders.py

```diff
@@ -43,14 +44,16 @@ def render_template(self, notification, recipient):
             },
         )
 
-        # "Force" rendering the whole template (including global variables).
-        # Since we render block by block afterwards, the context and variables
-        # would be lost between blocks.
-        list(template.root_render_func(ctx))
-
-        return {
-            block: "".join(
-                block_func(ctx)
-            )  # have to evaluate, as block_func is a generator
-            for block, block_func in template.blocks.items()
-        }
+        # Forcing the locale of the recipient so the correct language is chosen for translatable strings
+        with force_locale(locale):
```

Comment:
Great observation and question! It will raise a `babel.core.UnknownLocaleError`
So will have to ensure it is in the available locales as in the [selector](https://github.com/inveniosoftware/invenio-i18n/blob/7bfee5b7e78047c278c4621430f5d72e662d2c0b/invenio_i18n/selectors.py#L37-L39) or use the [method from the extension](https://github.com/inveniosoftware/invenio-i18n/blob/7bfee5b7e78047c278c4621430f5d72e662d2c0b/invenio_i18n/ext.py#L164-L175). We could also use this in the selector. Then would probably also add a proxy for the `i18n` extension (as with other modules), so it is easier to access.

What do you think?

Reply:
Yeah, I also noticed there's no `current_i18n`... It should be ok to add, but adds a bit of friction to get this change in (we would need to release `invenio-i18n` beforehand. See my comment below, in which case we might be able to get away with using something from `flask_babel`